bids/layout/layout.py

```diff
@@ -299,12 +299,30 @@ def __getattr__(self, key):
 
     def __repr__(self):
         """Provide a tidy summary of key properties."""
-        # TODO: Replace each nested list comprehension with a single DB query
-        n_sessions = len([session for isub in self.get_subjects()
-                          for session in self.get_sessions(subject=isub)])
-        n_runs = len([run for isub in self.get_subjects()
-                      for run in self.get_runs(subject=isub)])
-        n_subjects = len(self.get_subjects())
+        n_subjects = len(
+            [s.value
+             for s in self.session.query(Tag).filter_by(
```

Comment:
We call `self.session.query(Tag)` several times. does it make sense to start above with

```Python
tag_query = self.session.query(Tag)
```

And

```suggestion
             for s in tag_query.filter_by(
```

Or is this something where the work is only done when you try to iterate over the resulting object?